cvpy/tests/test_cvat_project.py

```diff
@@ -149,62 +158,74 @@ def test_cvat_project_post_images(self):
-        cas_connection.loadactionset('image')
-        cas_connection.addcaslib(name='dlib',
-                                 activeOnAdd=False,
-                                 path=TestCVATProject.datapath,
-                                 dataSource='PATH',
-                                 subdirectories=True)
-
-        cas_table_encoded = cas_connection.CASTable('cas_table_encoded')
-        cas_connection.image.loadimages(labellevels=5,
-                                        casout=cas_table_encoded,
-                                        caslib='dlib',
-                                        decode=False)
-
-        cas_connection.addcaslib(name='bigdata',
-                                 activeOnAdd=False,
-                                 path='/bigdisk/lax/patela/data',
-                                 dataSource='PATH',
-                                 subdirectories=True)
+        cas_table_encoded = self.cas_connection.CASTable('cas_table_encoded_save_test')
+        self.cas_connection.image.loadimages(path='images',
+                                             labellevels=5,
+                                             casout=cas_table_encoded,
+                                             caslib=self.caslib_name,
+                                             decode=False)
 
         # Create a CVATProject.
         url = TestCVATProject.cvat_url
 
-        project_name = 'MyDemoProject'
+        project_name = 'MyDemoProject_SaveTest'
         annotation_type = AnnotationType.OBJECT_DETECTION
 
         mountain_label = AnnotationLabel(name='Mountain', color='orange')
         person_label = AnnotationLabel(name='Person', color='green')
 
         labels = [mountain_label, person_label]
 
-        credentials = Credentials(self.cvat_username, self.cvat_password)
-        cvat_project = CVATProject(url=url, cas_connection=cas_connection, credentials=credentials,
+        credentials = Credentials(TestCVATProject.cvat_username, TestCVATProject.cvat_password)
+        cvat_project = CVATProject(url=url, cas_connection=self.cas_connection, credentials=credentials,
                                    project_name=project_name, annotation_type=annotation_type,
                                    labels=labels)
 
         # Post the images to the CVATProject.
         image_table_encoded = ImageTable(cas_table_encoded)
         cvat_project.post_images(image_table_encoded)
 
-        # Note: CVAT will throw an internal server error exception if we attempt to access the tasks
-        #       data to soon after an upload. So we have a sleep here to prevent the issue.
-        time.sleep(5)
+        # Save the project
+        cvat_project.save(self.caslib_name, 'cvpy', replace=True)
+
+    def test_cvat_project_resume(self):
+
+        # Resume a test project
+        cvat_project = CVATProject.resume(project_name='MyDemoProject', cas_connection=self.cas_connection,
```

Comment:
The project name here is "MyDemoProject", but in the save test it is "MyDemoProject_SaveTest". I assuming you're using the save test to create the table in the CAS session that will be used for the resume test; if that's the case, I'm not sure I see how this test works. Shouldn't it fail to find the correct table because the names are different?

Reply:
My intent is to use separate sashdat files for the save and resume tests since, in future, we might run these tests in parallel. So I have created a MyDemoProject.sashdat file for use by the resume function which will never be modified. The save function will overwrite the MyDemoProject_SaveTest.sashdat file each time it executes.